### DISCUSSION TITLE
How can i generate a image with round radius like this?

### DISCUSSION THREAD

**Discussion by jonny77** [libvips/php-vips#136] [Q&A]
orig image:
![image](https://user-images.githubusercontent.com/1962170/157214562-3db8c425-138e-4d0c-99a7-c8c63166564c.png)

wanted image:
![822481f32d15cde3ef857d720b31d4c9](https://user-images.githubusercontent.com/1962170/157214882-85245ef6-1aa5-4054-a22b-f1d9d97cf7ac.png)

**Comment by jcupitt** [comment]
Hello @jonny77,

The simplest way is to mask with an SVG. See https://github.com/libvips/libvips/discussions/2271

**Reply by jonny77** [reply]
> Hello @jonny77,
> 
> The simplest way is to mask with an SVG. See [libvips/libvips#2271](https://github.com/libvips/libvips/discussions/2271)

Thanks a lot ,   actually i have just png / jpg file . and i run vips in laravel . https://github.com/libvips/libvips/discussions/2271 is python

**Reply by jcupitt** [reply]
It's the same in php, eg.:

```php
#!/usr/bin/env php
<?php

require __DIR__ . '/vendor/autoload.php';
use Jcupitt\Vips;

$image = Vips\Image::newFromFile($argv[1], ['access' => 'sequential']);
    
$cx = $image->width / 2;
$cy = $image->height / 2;
$r = min($cx, $cy);

$svg = <<<EOF
<svg viewBox="0 0 {$image->width} {$image->height}">
    <circle cx="{$cx}" cy="{$cy}" r="{$r}" fill="#000" />
</svg>
EOF;

$mask = Vips\Image::svgload_buffer($svg);

# svgload always makes RGBA ... we want the alpha
$alpha = $mask[3];

$image->bandjoin($alpha)->writeToFile($argv[2]);
```

Then using this test image:

![240](https://user-images.githubusercontent.com/580843/157233526-28bf7ecb-3b09-4ac9-8fc1-393ef21c3ccb.jpg)


I can run:

```
$ ./crop-svg.php ~/pics/240.jpg x.png
```

To make:

![x](https://user-images.githubusercontent.com/580843/157233474-76db6fff-cafa-418d-9f1e-bbf8400a24b9.png)

**Reply by jonny77** [reply]
i tried so , and i got the right image what i want , though it not circle . 
 i want to generate a 300x300 transparent image , should i run this?
 * @method static Image black(integer $width, integer $height, array $options = []) Make a black image.

but i do not know how to specify the options value , i checked the docs , but see nothing about the $option

**Reply by jcupitt** [reply]
Something like:

```php
$black = Vips\Image::black(300, 300, [
    'bands' => 4
]);
```
